### PR TITLE
fix: APIC-429 not rendering response body when is array and items are…

### PR DESCRIPTION
**Pull request by carowright**
… union of type and properties

Bug: Empty body response when its defined as an array of another type and extra properties.
ResponseBodyTypeNotRendering

Fix: When type changes and is of type array compute _and types_ as well

### CODE REVIEW

File: src/ApiTypeDocument.js [hunk 1]
```diff
@@ -298,6 +298,17 @@ export class ApiTypeDocument extends PropertyDocumentMixin(LitElement) {
...
     ) {
       isArray = true;
+
+      const iKey = this._getAmfKey(this.ns.aml.vocabularies.shapes.items);
+      let items = this._ensureArray(type[iKey]);
+      if (items) {
+        items = items[0];
+        const key = this._getAmfKey(this.ns.w3.shacl.and);
+        if (key in items) {
+          isAnd = true;
```
Comment (jarrodek) [inline src/ApiTypeDocument.js:308]: Should it still be marked as isArray in this case?

File: test/api-type-document.test.js [hunk 2]
```diff
@@ -34,6 +42,54 @@ describe('<api-type-document>', function() {
     return [schema, mt];
   }
 
+  function getResponseSchema(element, model, path, methodName, statusCode) {
```
Comment (jarrodek) [inline test/api-type-document.test.js:45]: For these functions I was usually using AmfLoader to keep it there. You can easily use it later if needed.

File: test/api-type-document.test.js [hunk 3]
```diff
@@ -34,6 +42,54 @@ describe('<api-type-document>', function() {
...
+        m,
+        element.ns.aml.vocabularies.apiContract.method
+      );
+      if (value === methodName) {
+        method = m;
+        break;
+      }
+    }
+
+    const rKey = element._getAmfKey(
```
Comment (jarrodek) [inline test/api-type-document.test.js:66]: Until up to this like you can use `AmfLoader.lookupOperation` function.

File: test/api-type-document.test.js [hunk 4]
```diff
@@ -229,6 +285,22 @@ describe('<api-type-document>', function() {
...
+          element.amf = data[0];
+          const payload = getResponseSchema(
+            element,
+            data[0],
+            '/pets',
+            'get',
+            '200'
+          );
+          element._typeChanged(element._resolve(payload[0]));
+          assert.isTrue(element.isArray);
```
Comment (jarrodek) [inline test/api-type-document.test.js:300]: I believe this is incorrect. The `isArray` renders the template for the array view. It shows nothing now but the template is still in the DOM. Should be removed to reduce proc/mem usage. This should also test whether `isAnd` is set.

File: src/ApiTypeDocument.js (hunk 1 above)
Comment (carowright) [reply src/ApiTypeDocument.js:309]: Updating isArray to false for this case

**Pull request merged**